mesonbuild/backend/backends.py

```diff
@@ -429,6 +429,9 @@ def generate_basic_compiler_args(self, target, compiler, no_warn_args=False):
         # NOTE: We must preserve the order in which external deps are
         # specified, so we reverse the list before iterating over it.
         for dep in reversed(target.get_external_deps()):
+            if not dep.found():
+                continue
+
```

Comment:
It doesn't matter either way because other not-found dependencies have empty link args and compile args.